tests/client/test_client_tenant_token.py

```diff
@@ -0,0 +1,94 @@
+# pylint: disable=invalid-name
+
+from re import search
+import pytest
+import meilisearch
+from tests import BASE_URL, MASTER_KEY
+from meilisearch.errors import MeiliSearchApiError
+import datetime
+
+def test_generate_tenant_token_with_search_rules(get_private_key, index_with_documents):
+    """Tests create a tenant token with only search rules."""
+    index_with_documents()
+    key = get_private_key
+
+    client = meilisearch.Client(BASE_URL, key['key'])
+
+    token = client.generate_tenant_token(search_rules=["*"])
+
+    token_client = meilisearch.Client(BASE_URL, token)
+    response = token_client.index('indexUID').search('')
+    assert isinstance(response, dict)
+    assert len(response['hits']) == 20
```

Comment:
Actually, you don't really care about the size of the response, because it's not related to the key generation itself. But you do care if the `/search` request answer with a success status!

Reply:
I'm not sure to understand what you mean. There is no `status` field in the search response so I just check if the response has enough results to be valid

Reply:
What I want to mean is, if your request returned just 10 records caused by a new version of the engine v0.27 for example, **that's an error? this test should break because of that change?** In this case no, because your test just wanted to check if your generated token can make a request successfully.

But, since the assertion is `check if the returned records are equal to 20` now you have to fix this test too because the core engine changed internally.

Giving more context: this idea does not apply everywhere, because there are some use-cases we will need to handle that, ie. the tests regarding the filters or the search itself.

A different approach is to use more unit testing for some cases instead of e2e tests give a look here:

- https://www.onpathtesting.com/blog/end-to-end-vs-integration-testing
- https://prodperfect.com/blog/test-development/end-to-end-or-unit-testing-which-tests-for-which-bugs/

Reply:
I get it you mean if the search `limit` by default is modified from 20 to 10.
I can change the request with a specific limit like:
```
    response = token_client.index('indexUID').search('', {
        'limit': 5
    })
```
This way I'm sure the search works well and the test will not be breaking. What do you think?